source/common/http/utility.h

```diff
@@ -329,6 +329,9 @@ struct LocalReplyData {
   const absl::optional<Grpc::Status::GrpcStatus> grpc_status_;
   // Tells if this is a response to a HEAD request.
   bool is_head_request_ = false;
+  // Tells whether the localReply HTTP status code for gRPC calls is taken from
+  // the called service's response headers.
```

Comment:
I think these code comments and even the nature of this boolean are a bit confusing. What is the "called service"? This makes sense in ext_authz, when we make a RPC sidecall. What does this mean for other filters etc.

The main concern here is that we are leaking ext_authz details into Envoy core.

Reply:
Would you prefer to have it say "whether to retain HTTP status in response headers for gRPC calls" and remove the ambiguous/wordy "called service"?  

It's sending an ext_authz decision back into Envoy core, by way of the boolean.  Other filters might leverage this decision-passing mechanism differently, but I agree that it's constrained by having been designed for this ext_authz use case.

Reply:
I think it might be clearer to describe it in terms of other features of this options struct. Namely `if is_grpc_ and retain_http_status_for_grpc_ is set, then the HTTP status code will be response_code_ instead of the gRPC standard 200`